src/pages/studyView/charts/barChart/BarChartToolTip.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import * as _ from 'lodash';
+import {computed} from "mobx";
+import { Popover } from "react-bootstrap";
+import classnames from "classnames";
+import { formatRange } from "pages/studyView/StudyViewUtils";
+
+export type ToolTipRange = {
+    start: number | undefined,
+    end: number | undefined,
+    special: string | undefined
+};
+
+export type BarChartToolTipProps = {
+    mousePosition: { x: number, y: number },
+    totalBars: number,
+    currentBarIndex: number,
+    range: ToolTipRange | null,
+    sampleCount: number,
+    windowWidth: number,
+};
+
+export const VERTICAL_OFFSET = 17;
+export const HORIZONTAL_OFFSET = 8;
+
+const WIDTH = 150;
+
+export default class BarChartToolTip extends React.Component<BarChartToolTipProps, {}> {
```

Comment:
This might need to be @observer depending on whether its props can change during lifecycle (mouse position)?